lib/koans/19_streams.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Streams do
+  use Koans
+
+  @intro "Streams"
+
+  koan "Streams are lazier than Enum, they only return a struct" do
+    assert Stream.map([1, 2, 3], &(&1 + 1)).__struct__ == ___
+  end
+
+  koan "You have to ask it explicitly to do the operation" do
+    updated_list =
+      [1, 2, 3]
+      |> Stream.map(&(&1 + 1))
```

Comment:
One way I can think of this is to play with side effects by having the individual 'transformations' that get stacked send a value to a pid and then inspecting the order in which they were received?